src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -116,8 +116,13 @@ object DemuxFastqs {
     Sample(sampleOrdinal=sampleOrdinal, sampleId=UnmatchedSampleId, sampleName=UnmatchedSampleId, libraryId=UnmatchedSampleId, i7IndexBases=Some(noMatchBarcode))
   }
 
-  private[fastq] def toSampleOutputPrefix(sample: Sample, isUnmatched: Boolean, output: DirPath, unmatched: String): PathPrefix = {
-    if (isUnmatched) PathUtil.removeExtension(output.resolve(unmatched)) else outputPrefixFrom(output, sample)
+  private[fastq] def toSampleOutputPrefix(sample: Sample, isUnmatched: Boolean, illuminaStandards: Boolean, output: DirPath, unmatched: String): PathPrefix = {
+    (isUnmatched, illuminaStandards) match {
+      case (true, true)   => output.resolve(f"${UnmatchedSampleId}_S${sample.sampleOrdinal}_L${sample.lane.getOrElse(1)}%03d")
```

Comment:
Just FYI, that different versions of Illumina software do different sets of character substitutions on the `sample.sampleName` part of this.  Some turn hyphens into underscores, some turn underscores into hyphens.  I think all replace spaces (with either underscores or hyphens).  So just beware that this isn't a perfect match for any illumina software version in all cases.

Reply:
Awesome.